casper/src/main/scala/coop/rchain/casper/Casper.scala

```diff
@@ -218,6 +216,15 @@ sealed abstract class MultiParentCasperInstances {
         _runtime.close()
       }
 
+      def normalizedInitialFault(weights: Map[Validator, Int]): F[Float] =
+        (equivocationsTracker
+          .map(_.equivocator)
+          .toSet
```

Comment:
Why bother with `toSet` here? It is already a set, though a mutable one. Is the idea to make an immutable copy for thread-safety? If so I think `map` already results in a new object which is local to this function.

Reply:
Interesting

```
Set[(Int,Int)]((1,2),(2,2),(3,2)).map(_._2)
res1: scala.collection.immutable.Set[Int] = Set(2)
```

I guess I was expecting it to return `res1: scala.collection.immutable.List[Int] = List(2,2,2)` and so in order to deduplicate. Thanks for the catch.

Reply:
Actually I'm going to leave it in there because it makes it clear that we are trying to deduplicate.